### PR TITLE
Task4 app settings(l5)

### PR DESCRIPTION
**Pull request by wi9916**

### CODE REVIEW

File: Web/BLL/RockFood/Services/FoodService.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ public List<IFoodable> Get()
 
         public IFoodable Get(int id)
         {
-            var food = _db.Foods.Find(id);
+            var food =  _db.Foods.Find(id);
```
Comment (yevhenbilushko) [inline Web/BLL/RockFood/Services/FoodService.cs]: var food =  _db.Foods.Find(id);
->
var food = _db.Foods.Find(id);

one space is eniuth

Comment (yevhenbilushko) [reply Web/BLL/RockFood/Services/FoodService.cs]: is enough

File: Web/DAL/Entity.Data/DataContext.cs [hunk 2]
```diff
@@ -12,11 +13,16 @@ namespace Entity.Data
...
         public DbSet<Food> Foods { get; set; }
         public DbSet<Customer> Customers { get; set; }
+        public DataContext(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
         protected override void OnConfiguring(DbContextOptionsBuilder builder)
         {
-            builder.UseSqlServer("Data Source=DESKTOP-4H5PP4L; Integrated Security = true; Initial Catalog = WebRockFood");
+            builder.UseSqlServer(_configuration["BdConnectionStrings"]);
```
Comment (yevhenbilushko) [inline Web/DAL/Entity.Data/DataContext.cs]: use it in startup.cs

File: Web/RockFood.Api/Controllers/FoodMVCController.cs [hunk 3]
```diff
@@ -12,26 +15,43 @@ namespace RockFood.Api.Controllers
...
+            {
+                if(_env.IsDevelopment())
+                {
+                    food.About += "IsDevelopment";
+                }
+                if (_env.IsEnvironment("QA"))
+                {
+                    food.About += "IsQa";
+                }
+                if (_env.IsEnvironment("Production"))
```
Comment (yevhenbilushko) [inline Web/RockFood.Api/Controllers/FoodMVCController.cs]: _env.IsEnvironment("Production")
->
_env.IsProduction()

File: Web/RockFood.Api/Filter/MyExceptionFilter.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+    public class MyExceptionFilter : IExceptionFilter
+    {
+        private readonly ILogger<MyExceptionFilter> _logger;
+        public MyExceptionFilter(ILogger<MyExceptionFilter> logger)
+        {
+            _logger = logger;
+        }
+        public void OnException(ExceptionContext context)
+        {
+            _logger.LogInformation($"{DateTimeOffset.UtcNow} Error: {context.Exception}");
```
Comment (yevhenbilushko) [inline Web/RockFood.Api/Filter/MyExceptionFilter.cs]: LogInformation
->
LogError

Comment (yevhenbilushko) [reply Web/RockFood.Api/Filter/MyExceptionFilter.cs]: log info depends on environment variable value

**Comment by yevhenbilushko** [comment]
and add build check

**Comment by wi9916** [comment]
Task4 app settings(l5) build check #14: Pull request #28 synchronize by wi9916

File: Web/RockFood.Api/Startup.cs [hunk 5]
```diff
@@ -32,19 +33,24 @@ public Startup(IConfiguration configuration)
...
+            services.AddScoped<IFoodService, FoodService>();
+            services.AddSingleton<IFoodable, Food>();           
             services.AddSingleton<ILogger<MyExceptionFilter>, Logger<MyExceptionFilter>>();
             services.AddSingleton<ILogger<FoodActionFilter>, Logger<FoodActionFilter>>();
+
             services.AddControllers();
             services.AddControllersWithViews();
 
+            services.AddDbContext<DataContext>(
+                options => options.UseSqlServer(Configuration.GetConnectionString("BdConnectionStrings")));
```
Comment (yevhenbilushko) [inline Web/RockFood.Api/Startup.cs]: Db
Database

**Review by yevhenbilushko — DISMISSED** [review]

**Review by yevhenbilushko — APPROVED** [review]

**Pull request merged**